cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/ExternalDatasets.java

```diff
@@ -107,7 +108,7 @@ public static Output makeTrackable(Admin admin, Output output) {
 
     // Output is a dataset that implements input format provider,
     // this can be tracked by itself without creating an external dataset
-    if (outputFormatProvider instanceof Dataset) {
+    if (outputFormatProvider instanceof Dataset || outputFormatProvider instanceof NullOutputFormatProvider) {
```

Comment:
Since we are returning output as it is, can we not call this method when in preview mode?